### PR TITLE
Documentation implemented

### PR DESCRIPTION
**Pull request by 0ctagon**
Docstrings updated

Fully functional readthedocs: https://plothist.readthedocs.io/en/latest/index.html

### CODE REVIEW

**Comment by 0ctagon** [comment]
Also propose that when this is merged to main, we publish on PyPI under v0.1

File: docs/advanced/hep_examples.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        rotation=1.5,
+        darkest=0.3,
+        lightest=0.8,
+        reverse=True,
+    )
+    background_masks = [df[category] == p for p in background_categories]
+
+    # Make histograms
+
+    data_hist = make_hist(df[key][data_mask], bins=50, range=range, weights=1)
```
Comment (cyrraz) [inline docs/advanced/hep_examples.rst:38]: make_hist not defined

Comment (cyrraz) [reply docs/advanced/hep_examples.rst:38]: (I will not comment again for the other examples, but putting the import helps the user to just copy and paste)

File: docs/basics/2d_hist.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    from plothist.plotters import make_2d_hist, plot_2d_hist
+    from plothist.plotters import create_variable_registry, update_variable_registry_ranges, get_variable_from_registry
+
+    # No need to redo this step if the registry was already created before
+    variable_keys = ["variable_0", "variable_1", "variable_2"]
+    create_variable_registry(variable_keys)
+    update_variable_registry_ranges(df, variable_keys)
+
+    # Get all the correlation plot between the variables
+    variable_keys_combinations = list(itertools.combinations(variable_keys, 2))
```
Comment (cyrraz) [inline docs/basics/2d_hist.rst:55]: nice :)

File: docs/img/2d_hist_simple.png
Comment (cyrraz) [inline docs/img/2d_hist_simple.png:1]: png = :(

File: docs/img/intro_2dhist_complexe.svg
Comment (cyrraz) [inline docs/img/intro_2dhist_complexe.svg:1]: svg = :)

File: plothist/plotters.py [hunk 3]
```diff
@@ -63,6 +89,36 @@ def create_variable_registry(variables, path="./variable_registry.yaml", reset=F
 
 
 def get_variable_from_registry(variable, path="./variable_registry.yaml"):
+    """
+    This function retrieves the parameter information for a variable from the variable registry file specified by the 'path' parameter.
+    It loads the variable registry file and returns the dictionary entry corresponding to the specified variable name.
+
+    If the variable registry file does not exist, a `RuntimeError` is raised, unless the default path ("./variable_registry.yaml") is used.
+    In that case, it is assumed that the `create_variable_registry` function has not been executed yet.
```
Comment (cyrraz) [inline plothist/plotters.py:97]: Weird behavior. We should always raise a RuntimeError if the registry does not exist

File: plothist/plotters.py [hunk 4]
```diff
@@ -63,6 +89,36 @@ def create_variable_registry(variables, path="./variable_registry.yaml", reset=F
...
+
+    Returns
+    -------
+    dict
+        A dictionary containing the parameter information for the specified variable.
+
+    Raises
+    ------
+    RuntimeError
+        If the variable registry file does not exist, and the default path is used.
```
Comment (cyrraz) [inline plothist/plotters.py:115]: weird

File: plothist/plotters.py [hunk 5]
```diff
@@ -63,6 +89,36 @@ def create_variable_registry(variables, path="./variable_registry.yaml", reset=F
...
+
+    See also
+    --------
+    create_variable_registry
```
Comment (cyrraz) [inline plothist/plotters.py:119]: nice

File: plothist/plotters.py [hunk 6]
```diff
@@ -76,6 +132,33 @@ def update_variable_registry(
...
+        The path to the variable registry file (default is "./variable_registry.yaml").
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    RuntimeError
+        If the variable registry file does not exist, and the default path is used.
```
Comment (cyrraz) [inline plothist/plotters.py:156]: weird

File: plothist/plotters.py [hunk 7]
```diff
@@ -63,6 +89,36 @@ def create_variable_registry(variables, path="./variable_registry.yaml", reset=F
...
+
+    See also
+    --------
+    create_variable_registry
+    """
+
     if not os.path.exists(path):
         if path == "./variable_registry.yaml":
```
Comment (cyrraz) [inline plothist/plotters.py:123]: Weird behavior. We should always raise a RuntimeError if the registry does not exist

File: plothist/plotters.py [hunk 8]
```diff
@@ -76,6 +132,33 @@ def update_variable_registry(
...
+    ------
+    RuntimeError
+        If the variable registry file does not exist, and the default path is used.
+
+    See Also
+    --------
+    create_variable_registry
+    """
     if not os.path.exists(path):
         if path == "./variable_registry.yaml":
```
Comment (cyrraz) [inline plothist/plotters.py:163]: Weird behavior. We should always raise a RuntimeError if the registry does not exist

File: plothist/plotters.py [hunk 9]
```diff
@@ -153,19 +299,29 @@ def _flatten_2d_hist(hist):
...
-    binning : (int, float, float), default: (50,None,None)
-        Binning, with format (number of bins, lower bound, upper bound)
-    weights : float or array-like, default=1
-        A single weight to apply to all the data points, or an array-like to apply weights to each data point
+    data : array-like
+        1D array-like data used to fill the histogram.
+    bins : int or tuple, optional
+        Binning specification for the histogram (default is 50).
+        If an integer, it represents the number of bins.
+        If a tuple, it should be in the format (number of bins, lower bound, upper bound).
```
Comment (cyrraz) [inline plothist/plotters.py:312]: wrong

Comment (cyrraz) [reply plothist/plotters.py:312]: If a tuple, it is the explicit list of all bin edges

File: plothist/plotters.py [hunk 10]
```diff
@@ -190,19 +346,34 @@ def make_hist(data, bins=50, range=None, weights=1):
...
-        Data used to fill a histogram.
-    binning : list((int, float, float))
-        Binning, with format (number of bins, lower bound, upper bound)
-    weights : float or array-like, default=1
-        A single weight to apply to all the data points, or an array-like to apply weights to each data point
+    data : array-like
+        2D array-like data used to fill the histogram.
+    bins : tuple, optional
+        Binning specification for each dimension of the histogram (default is (10, 10)).
+        Each element of the tuple represents the number of bins for the corresponding dimension.
```
Comment (cyrraz) [inline plothist/plotters.py:358]: Also support explicit bin edges specification (for non-constant bin size)

File: plothist/plotters.py (hunk 6 above)
Comment (0ctagon) [reply plothist/plotters.py:156]: The idea is:

If the file does not exist, 2 scenarios:

1.The case specified in the documentation: the user forgot to run `create_var_reg()`, the `RuntimeError` (here in the doc) tells him to run this function
2. the user inputted an incorrect `yaml` path (so he knows what he do), so the `safe_load()` of the `yaml` returns a `RuntimeError` incorrect path

The documentation might be misleading, I thing removing this `RuntimeError` from the documentation is better.

File: plothist/plotters.py (hunk 7 above)
Comment (0ctagon) [reply plothist/plotters.py:123]: [same](https://github.com/cyrraz/plothist/pull/3#discussion_r1220610562)

File: plothist/plotters.py (hunk 8 above)
Comment (0ctagon) [reply plothist/plotters.py:163]: [same](https://github.com/cyrraz/plothist/pull/3#discussion_r1220610562)

**Comment by 0ctagon** [comment]
Small note: don't remove the docs branch, it will be the dev branch for the readthedoc (and the main pip version will be the official doc) (|Docs from latest|  and |Docs from main| buttons)

**Review by cyrraz — APPROVED** [review]

**Pull request merged**